services/QuillLMS/app/queries/staff/rules_analysis_query.rb

```diff
@@ -0,0 +1,104 @@
+        avg(DISTINCT CAST(JSON_VALUE(metadata, '$.api.confidence') AS FLOAT64)) AS avg_confidence,
+        count(DISTINCT CASE WHEN feedback_history_ratings.rating = true THEN feedback_history_ratings.id END) AS total_strong,
+        count(DISTINCT CASE WHEN feedback_history_ratings.rating = false THEN feedback_history_ratings.id END) AS total_weak,
+        count(DISTINCT CASE WHEN feedback_history_flags.flag = 'repeated-consecutive' THEN feedback_history_flags.id END) AS repeated_consecutive,
+        count(DISTINCT CASE WHEN feedback_history_flags.flag = 'repeated-non-consecutive' THEN feedback_history_flags.id END) AS repeated_non_consecutive
+      SQL
+    end
+
+    def from_and_join_clauses
+      <<-SQL
+        FROM lms.comprehension_rules
+        INNER JOIN lms.comprehension_prompts_rules as prompts_rules ON comprehension_rules.id = prompts_rules.rule_id
+        INNER JOIN lms.comprehension_prompts as prompts ON prompts_rules.prompt_id = prompts.id
+        LEFT JOIN lms.feedback_histories ON feedback_histories.rule_uid = comprehension_rules.uid AND feedback_histories.prompt_id = prompts.id
+        LEFT JOIN lms.feedback_history_ratings ON feedback_histories.id = feedback_history_ratings.feedback_history_id
+        LEFT JOIN lms.feedback_history_flags ON feedback_histories.id = feedback_history_flags.feedback_history_id
+      SQL
+    end
+
+    def start_date_where_clause
+      start_date ? "AND feedback_histories.time >= '#{DateTime.parse(start_date).to_fs(:db)}'" : ""
+    end
+
+    def end_date_where_clause
+      end_date ? "AND feedback_histories.time <= '#{DateTime.parse(end_date).to_fs(:db)}'" : ""
+    end
+
+    def activity_version_where_clause
+      activity_version ? "AND feedback_histories.activity_version = #{activity_version}" : ""
+    end
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    def post_query_transform(query_result)
+      rules_with_feedbacks = Evidence::Rule.where(id: query_result.pluck(:id)).includes(:feedbacks)
+      query_result.map do |r|
+        first_feedback, second_feedback = rules_with_feedbacks.find{|x| x.id == r[:id]}&.feedbacks&.sort_by(&:order)
+        {
+            rule_uid: r[:rules_uid],
+            api_name: r[:rule_type],
+            rule_order: r[:rule_suborder],
+            first_feedback: first_feedback&.text || '',
+            second_feedback: second_feedback&.text || '',
+            rule_note: r[:rule_note],
+            rule_name: r[:rule_name],
+            avg_confidence: r[:avg_confidence] ? (r[:avg_confidence] * 100).round : nil,
+            total_responses: r[:total_responses],
+            strong_responses: r[:total_strong],
+            weak_responses: r[:total_weak],
+            repeated_consecutive_responses: r[:repeated_consecutive],
+            repeated_non_consecutive_responses: r[:repeated_non_consecutive],
+        }
+      end
+      # rubocop:enable Metrics/CyclomaticComplexity
```

Comment:
I think we should avoid new code with these rubocop exceptions. Look to refactor this. Some suggestions above.